src/Sprites/Animation.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "../Point.h"
+#include "../Rect.h"
+#include <vector>
+#include <cstdint>
+
+struct Animation {
+	struct Frame {
+		struct SubFrame {
+			uint16_t bitmapIndex;
+			uint8_t unknown;
+			uint8_t frameIndex;
+			Point16 pixelOffset;
+		};
+
+		uint8_t unknown;
+		uint8_t optional1;
+		uint8_t optional2;
+		uint8_t optional3;
+		uint8_t optional4;
+
+		// Limited to size of uint16_t MAX
+		std::vector<SubFrame> subFrames;
```

Comment:
I assume  you mean the vector index is limited to fit within a `uint16_t`. I had to think about the meaning here, and make sure it wasn't about the vector occupying 0, 1, or 2 bytes only. Seeing the optional fields above put me in that mindset.

Reply:
Good catch on the bitfield controlled size. I forgot about the masking and the 7-bit bitfield part.